great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -509,6 +513,26 @@ def get_domain_records(
         if TextClause and isinstance(selectable, TextClause):
             selectable = selectable.columns().subquery()
 
+        filter_conditions: List[RowCondition] = domain_kwargs.get(
+            "filter_conditions", []
+        )
+        # For SqlAlchemyExecutionEngine only one filter condition is allowed
+        if len(filter_conditions) == 1:
+            filter_condition = filter_conditions[0]
+            assert (
+                filter_condition.type_ == RowConditionParserType.GE
+            ), "filter_condition must be of type GE for SqlAlchemyExecutionEngine"
+
+            selectable = (
+                sa.select([sa.text("*")])
+                .select_from(selectable)
+                .where(parse_condition_to_sqlalchemy(filter_condition.condition))
+            )
+        elif len(filter_conditions) > 1:
+            raise GreatExpectationsError(
+                "SqlAlchemyExecutionEngine currently only supports a single filter condition."
+            )
```

Comment:
If we only support a single condition, is there a reason we have a list here?

Reply:
Good point - in the future I'm hoping to pass a list of RowCondition objects to the execution engine to unify the interface across all execution engines. Figure this was a good starting step, but happy to remove if too forward looking.